offheap-resource/src/main/java/org/terracotta/offheapresource/PhysicalMemory.java

```
@@ -15,79 +15,70 @@
-        LOGGER.trace("Bean lookup failed on {}", s, e);
-      } catch (IllegalAccessException e) {
-        LOGGER.trace("Bean lookup failed on {}", s, e);
-      } catch (IllegalArgumentException e) {
-        LOGGER.trace("Bean lookup failed on {}", s, e);
-      } catch (InvocationTargetException e) {
-        LOGGER.trace("Bean lookup failed on {}", s, e);
+  private static Method find(String methodName) {
+    try {
+      Method method = ManagementFactory.getOperatingSystemMXBean().getClass().getMethod(methodName);
+      if (!method.isAccessible()) {
+        method.setAccessible(true);
       }
+      return method;
+    } catch (NoSuchMethodException e) {
+      LOGGER.warn("Unable to find method '{}' on the {}", methodName, OperatingSystemMXBean.class.getSimpleName());
+      return null;
     }
-    for (Class<?> i : OS_BEAN.getClass().getInterfaces()) {
-      try {
-        T result = (T) i.getMethod(name).invoke(OS_BEAN);
-        LOGGER.trace("Bean lookup successful using {}, got {}", i, result);
-        return result;
-      } catch (SecurityException e) {
-        LOGGER.trace("Bean lookup failed on {}", i, e);
-      } catch (NoSuchMethodException e) {
-        LOGGER.trace("Bean lookup failed on {}", i, e);
-      } catch (IllegalAccessException e) {
-        LOGGER.trace("Bean lookup failed on {}", i, e);
-      } catch (IllegalArgumentException e) {
-        LOGGER.trace("Bean lookup failed on {}", i, e);
-      } catch (InvocationTargetException e) {
-        LOGGER.trace("Bean lookup failed on {}", i, e);
+  }
+
+  private static Long invoke(Method method) {
+    try {
+      return method == null ? null : (Long) method.invoke(ManagementFactory.getOperatingSystemMXBean());
+    } catch (Throwable throwable) {
+      if (throwable instanceof Error) {
+        throw (Error) throwable;
+      }
+      if (throwable instanceof RuntimeException) {
+        throw (RuntimeException) throwable;
       }
+      throw new RuntimeException(throwable);
```

Why are we throwing exceptions rather than just returning null here?